src/kili/adapters/kili_api_gateway/asset/operations_mixin.py

```diff
@@ -34,13 +36,13 @@ def list_assets(
         options: QueryOptions,
     ) -> Generator[Dict, None, None]:
         """List assets with given options."""
-        added_label_id_field = added_latest_label_id_field = False
-        if "labels.jsonResponse" in fields and "labels.id" not in fields:
-            fields = (*fields, "labels.id")
-            added_label_id_field = True
-        if "latestLabel.jsonResponse" in fields and "latestLabel.id" not in fields:
-            fields = (*fields, "latestLabel.id")
-            added_latest_label_id_field = True
+        if "labels.jsonResponse" in fields or "latestLabel.jsonResponse" in fields:
+            # Check if we can get the jsonResponse of if we need to rebuild it.
+            project_info = get_project(
+                self.graphql_client, filters.project_id, ("inputType", "jsonInterface")
+            )
+            if project_info["inputType"] in {"VIDEO", "LLM_RLHF"}:
+                yield from self.list_assets_split(filters, fields, options, project_info)
```

Comment:
Here you should put a return after to prevent from passing to the next part once these iterations are finished